packages/cli/src/metadataGeneration/typeResolver.ts

```diff
@@ -152,143 +156,224 @@ export class TypeResolver {
-          const declaration = getDeclaration(property) as ts.PropertySignature | ts.PropertyDeclaration | ts.ParameterDeclaration | undefined;
-
-          if (declaration && ts.isPropertySignature(declaration)) {
-            return { ...this.propertyFromSignature(declaration, mappedTypeNode.questionToken), name: property.getName() };
-          } else if (declaration && (ts.isPropertyDeclaration(declaration) || ts.isParameter(declaration))) {
-            return { ...this.propertyFromDeclaration(declaration, mappedTypeNode.questionToken), name: property.getName() };
-          }
-
-          // Resolve default value, required and typeNode
-          let required = false;
-          const typeNode = this.current.typeChecker.typeToTypeNode(propertyType, undefined, ts.NodeBuilderFlags.NoTruncation)!;
-          if (mappedTypeNode.questionToken && mappedTypeNode.questionToken.kind === ts.SyntaxKind.MinusToken) {
-            required = true;
-          } else if (mappedTypeNode.questionToken && mappedTypeNode.questionToken.kind === ts.SyntaxKind.QuestionToken) {
-            required = false;
-          }
 
-          // Push property
+      const calcMappedType = (type: ts.Type): Tsoa.Type => {
+        if (type.flags & ts.TypeFlags.Union) {
+          //Intersections are not interesting somehow...
+          const types = (type as ts.UnionType).types;
+          const resolvedTypes = types.map(calcMappedType);
           return {
-            name: property.getName(),
-            required,
-            // Mapped types with any amount of indirection (template strings, unions, Exclude<>, etc.)
-            // don't provide an underlying declaration in the AST, thus we cannot know if the original
-            // property is deprecated. This matches intellisense's behavior in vscode.
-            deprecated: false,
-            type: new TypeResolver(typeNode, this.current, this.typeNode, this.context, this.referencer).resolve(),
-            validators: {},
+            dataType: 'union',
+            types: resolvedTypes,
           };
-        });
-
-      const objectLiteral: Tsoa.NestedObjectLiteralType = {
-        dataType: 'nestedObjectLiteral',
-        properties,
+        } else if (type.flags & ts.TypeFlags.Object) {
+          const typeProperties: ts.Symbol[] = type.getProperties();
+          const properties: Tsoa.Property[] = typeProperties
+            // Ignore methods, getter, setter and @ignored props
+            .filter(property => isIgnored(property) === false)
+            // Transform to property
+            .map(property => {
+              const propertyType = this.current.typeChecker.getTypeOfSymbol(property);
+
+              const typeNode = this.current.typeChecker.typeToTypeNode(propertyType, undefined, ts.NodeBuilderFlags.NoTruncation)!;
+              const parent = getOneOrigDeclaration(property); //If there are more declarations, we need to get one of them, from where we want to recognize jsDoc
```

Comment:
If there are more than one declaration for a type member, then choose one of them for handling jsdoc.

Reply:
In all other cases we throw if there are multiple. This seems like an intentional deviation?

Reply:
Currently, tsoa gets the [0]-th element. I think my code is not worse than the original.
Of course, I can make, that the parent member of TypeResolver is not a single value, but an array, and everywhere we want to use it, we check if every elements gets the same result or not. If you think it makes a better world.